### PR TITLE
Wrap specific padding behavior for convolution layer in tf.keras

### PR DESCRIPTION
**Pull request by Zhaopudark**
# Description

more details in #2674

Extend padding behavior of a keras convolution layer.
For example, a user can get a convolution layer with "same" padding in shape-wise and "reflect" padding in numerical-wise simultaneously. Such as:
```
x = tf.constant([1.,2.,3.,4.,5.],shape=[1,5,1])
conv1d_ = tf.keras.layers.Conv1D(filters=1, kernel_size=[2,]*1, strides=(1,)*1, padding="same")
conv1d = SpecificConvPad(conv1d_, padding_mode='constant',padding_constant=1)
y = conv1d(x)
````

## Type of change
- [x] Additional Testing
- [x] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [x] By running pre-commit hooks
- [x] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?
Shape-wise padding behavior has been tested to ensure to be consistent with original padding behavior.

### CODE REVIEW

**Comment by bhack** [comment]
Thanks for the proposal, can you try to open a ticket and propose this in Keras directly:
https://github.com/keras-team/keras

**Comment by bhack** [comment]
I think you need to contribute/propose this in Keras or Keras-cv repos
